tests/integration/src/incredible_dot_product.rs

```diff
@@ -0,0 +1,178 @@
+                            alloy::providers::fillers::NonceFiller,
+                            alloy::providers::fillers::ChainIdFiller,
+                        >,
+                    >,
+                >,
+            >,
+            alloy::providers::fillers::WalletFiller<EthereumWallet>,
+        >,
+        alloy::providers::RootProvider,
+    >,
+>;
+
+#[tokio::test]
+async fn test_incredible_dot_product() {
+    let (_container, http_endpoint, ws_endpoint) =
+        start_anvil_with_state(INCREDIBLE_DOT_PRODUCT_STATE_PATH).await;
+
+    init_logger(LogLevel::Info);
+    let logger = get_test_logger();
+
+    // Response calculator
+    let response_calculator = || response_calculator_from_fn(dot_product);
+
+    // Task spammer should finish when all tasks are created (`NUM_TASKS` * `TASK_INTERVAL`)
+    // so we add 5 seconds to the timeout
+    let timeout_duration = Duration::from_secs(NUM_TASKS * TASK_INTERVAL + 5);
+
+    // Create the AVS config using defaults
+    let config = AvsConfig::with_default_addresses_and_keys(
+        create_task_manager_contract(&http_endpoint, AGGREGATOR_SIGNER).await,
+        create_task_manager_contract(&http_endpoint, CHALLENGER_SIGNER).await,
+        create_task_manager_contract(&http_endpoint, TASK_SPAMMER_SIGNER).await,
+        response_calculator,
+        generate_input,
+        logger,
+        timeout_duration,
+        http_endpoint.to_string(),
+        ws_endpoint.to_string(),
+        AGGREGATOR_RPC_URL.to_string(),
+        TIME_TO_EXPIRY,
+        WINDOW_DURATION,
+        TASK_INTERVAL,
+        QUORUM_THRESHOLD,
+        QUORUMS.to_vec(),
+        NUM_TASKS,
+        DEPOSIT_TOKENS.to_string(),
+        NEW_MAGNITUDE.to_vec(),
+    );
+
+    // Start the AVS
+    start_avs(&config).await;
+
+    // Give some time to the aggregator to process the last task
+    tokio::time::sleep(Duration::from_secs(TASK_INTERVAL)).await;
+
+    verify_tasks_completed(&http_endpoint).await;
+}
+
+/// Verify that all tasks created by the task spammer have been completed
+async fn verify_tasks_completed(http_endpoint: &str) {
+    let contract = create_task_manager_contract(http_endpoint, AGGREGATOR_SIGNER).await;
+    let latest_task_num = contract.latestTaskNum().call().await.unwrap()._0;
+    assert_eq!(latest_task_num, NUM_TASKS as u32);
+
+    for task_index in 0..latest_task_num {
+        let response_hash = contract
+            .allTaskResponses(task_index)
+            .call()
+            .await
+            .unwrap()
+            ._0;
+        assert_ne!(B256::default(), response_hash);
+    }
+}
+
+/// Create the task manager contract with a specific signer
+async fn create_task_manager_contract(http_endpoint: &str, signer: &str) -> TaskManagerInstance {
+    let task_manager_address = Address::from_str(TASK_MANAGER_ADDRESS).unwrap();
+    let provider = get_signer(signer, http_endpoint);
+    IncredibleDotProductTaskManagerInstance::new(task_manager_address, provider)
+}
```

Comment:
I think it's easier to inline this so we don't have to declare `TaskManagerInstance`